protocols/luci-proto-sstp/htdocs/luci-static/resources/protocol/sstp.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+'require form';
+'require network';
+
+network.registerPatternVirtual(/^sstp-.+$/);
+
+return network.registerProtocol('sstp', {
+	getI18n: function() {
+		return _('SSTP');
+	},
+
+	getIfname: function() {
+		return this._ubus('l3_device') || 'sstp-%s'.format(this.sid);
+	},
+
+	getOpkgPackage: function() {
+		return 'sstp-client';
+	},
+
+	isFloating: function() {
+		return true;
+	},
+
+	isVirtual: function() {
+		return true;
+	},
+
+	getDevices: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var dev = this.getL3Device() || this.getDevice(), o;
+
+		// -- general ---------------------------------------------------------------------
+
+		o = s.taboption('general', form.Value, 'server', _('SSTP Server'));
+		o.datatype = 'host';
+
+		o = s.taboption('general', form.Value, 'username', _('PAP/CHAP username'));
+
+		o = s.taboption('general', form.Value, 'password', _('PAP/CHAP password'));
+		o.password = true;
+
+		// -- advanced --------------------------------------------------------------------
+
+		o = s.taboption('advanced', form.Flag, 'ipv6', _('IPv6 support'), _('If checked, adds "+ipv6" to the pppd options'));
+
+		o = s.taboption('advanced', form.ListValue, 'log_level', _('sstpc log-level'));
```

Comment:
Two small nit-picks:
 - Can you change the label to just `Log-level` (w/ capitalized L for consistency with other labels)
 - Add translation context to the number labels: `_('0', 'sstp log level value')` - the second argument won't be visible to the user but serve as hint for translators

Reply:
Does this need to be translated at all? Having something like "o.value('0');" also seems to work, but I haven't seen any explicit mention of this approach in the official documentation, so I'm avoiding it in case this would break things in the future.